subprojects/execution/src/main/java/org/gradle/internal/execution/model/InputNormalizer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.model;
+
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.internal.fingerprint.Normalizer;
+
+// TODO Break this up between simple normalizers and Java classpath normalizers
+//      The latter should be moved to :normalization-java
+public enum InputNormalizer implements Normalizer {
```

Comment:
Do we need interface or we could threat InputNormalizer and OutputNormalizer as two separate things?

Reply:
I'd like to split up `InputNormalizer` into separate types, it's like this now because it's simpler to implement the serialization in `TaskNodeCodec`.